tests/cpp/test_scatter.cpp

```diff
@@ -26,107 +26,247 @@ namespace nvfuser {
 class ScatterTest : public NVFuserTest {
  protected:
   void SetUp() override {
-    // To make the tests using std::rand deterministic
-    std::srand(0);
+    NVFuserTest::SetUp();
+    EnableOptionsGuard::getCurOptions().set(EnableOption::IdModel, {"all"});
   }
 };
 
-namespace {
-auto randomVector(int64_t low, int64_t high, int rank) {
-  std::vector<int64_t> out(rank, 0);
-  for (int idim = 0; idim < rank; ++idim) {
-    out[idim] = (std::rand() % (high - low)) + low;
+// Counting of non-duplicated integers on gmem with TIDx
+TEST_F(ScatterTest, BlockCountingWithGmem) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  const int64_t m = 64;
+  const int64_t n = 5;
+
+  auto tv0 = makeContigConcreteTensor({n}, DataType::Int);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+  auto tv2 = zeros({IrBuilder::create<Val>(m)}, DataType::Int);
+  auto tv3 = ones({IrBuilder::create<Val>(n)}, DataType::Int);
+  auto tv4 = scatter(tv2, 0, tv1, tv3);
+  fusion.addOutput(tv4);
+
+  for (auto tv : fusion.allTvs()) {
+    tv->axis(0)->parallelize(ParallelType::TIDx);
   }
-  return out;
+
+  // Scatter input must use the same memory as the output
+  tv2->setMemoryType(MemoryType::Global);
```

Comment:
looks like this is supposed to be part of manual scheduling?

Reply:
Yes, this is part of manual scheduling. Note that this test just exercises simple manual scheduling.